src/extension/background.js

```diff
@@ -59,6 +59,11 @@ chrome.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
   if (tab && tab.url && typeof internalActions[actionFromTab] === 'function') {
     resp = await internalActions[actionFromTab](message, sender);
   }
+
+  if (actionFromTab === 'updateUI') {
```

Comment:
...but instead add a `chrome.storage.local.onChange` listener and listen for changes to `display`.